src/quat.js

```diff
@@ -447,32 +447,75 @@ export function fromMat3(out, m) {
 }
 
 /**
- * Creates a quaternion from the given euler angle x, y, z.
+ * Creates a quaternion from the given euler angle x, y, z using the provided extrinsic order for the conversion.
  *
  * @param {quat} out the receiving quaternion
  * @param {x} Angle to rotate around X axis in degrees.
- * @param {y} Angle to rotate around Y axis in degrees.
  * @param {z} Angle to rotate around Z axis in degrees.
+ * @param {y} Angle to rotate around Y axis in degrees.
+ * @param {order} Intrinsic order for conversion, so far only 'xyz' and 'zyx' is supported.
  * @returns {quat} out
  * @function
  */
-export function fromEuler(out, x, y, z) {
+export function fromEuler(out, x, y, z, order='zyx') {
   let halfToRad = (0.5 * Math.PI) / 180.0;
   x *= halfToRad;
-  y *= halfToRad;
   z *= halfToRad;
+  y *= halfToRad;
 
   let sx = Math.sin(x);
   let cx = Math.cos(x);
-  let sy = Math.sin(y);
-  let cy = Math.cos(y);
   let sz = Math.sin(z);
   let cz = Math.cos(z);
+  let sy = Math.sin(y);
+  let cy = Math.cos(y);
```

Comment:
nit: the variable order has been needlessly rearranged and no longer matches the argument order -- just something I noticed!

Reply:
Done.